src/mlpack/methods/lsh/lsh_search_impl.hpp

```diff
@@ -331,12 +333,424 @@ void LSHSearch<SortPolicy>::BaseCase(const size_t queryIndex,
+  // neighbors are most likely to be. Likelihood is dependent only on a bin's
+  // score, which is the sum of scores of all dimension-action pairs, so we
+  // need to calculate the T smallest sums of scores that are not conflicting.
+  //
+  // Method:
+  // Store each perturbation set (pair of (dimension, action)) in a
+  // std::vector. Create a minheap of scores, with each node pointing to its
+  // relevant perturbation set. Each perturbation set popped from the minheap
+  // is the next most likely perturbation set.
+  // Transform perturbation set to perturbation vector by setting the
+  // dimensions specified by the set to queryCode+action (action is {-1, 1}).
+
+  /*
+  std::vector<size_t> Ao;
+  Ao.push_back(0); // initial perturbation holds smallest score (0 if sorted)
+
+  std::vector< std::vector<size_t> > perturbationSets;
+  perturbationSets.push_back(Ao); // storage of perturbation sets
+
+  std::priority_queue<
+    std::pair<double, size_t>,        // contents: pairs of (score, index)
+    std::vector<                      // container: vector of pairs
+      std::pair<double, size_t>
+      >,
+    std::greater< std::pair<double, size_t> > // comparator of pairs(compare scores)
+  > minHeap; // our minheap
+
+  // Start by adding the lowest scoring set to the minheap
+  // std::pair<double, size_t> pair0( perturbationScore(Ao, scores), 0 );
+  // minHeap.push(pair0);
+  minHeap.push( std::make_pair(perturbationScore(Ao, scores), 0) );
+
+  // loop invariable: after pvec iterations, additionalProbingBins contains pvec
+  // valid codes of the lowest-scoring bins (bins most likely to contain
+  // neighbors of the query).
+  for (size_t pvec = 0; pvec < T; ++pvec)
+  {
+    std::vector<size_t> Ai;
+    do
+    {
+      // get the perturbation set corresponding to the minimum score
+      Ai = perturbationSets[ minHeap.top().second ];
+      minHeap.pop(); // .top() returns, .pop() removes
+
+      // modify Ai (shift)
+      std::vector<size_t> As = Ai;
+      perturbationShift(As);
+      if ( perturbationValid(As, numProj) )
+      {
+        perturbationSets.push_back(As); // add shifted set to sets
+        minHeap.push( 
+            std::make_pair(
+              perturbationScore(As, scores), 
+              perturbationSets.size() - 1)
+            );
+      }
+
+      // modify Ai (expand)
+      std::vector<size_t> Ae = Ai;
+      perturbationExpand(Ae);
+      if ( perturbationValid(Ae, numProj) )
+      {
+        perturbationSets.push_back(Ae); // add expanded set to sets
+        minHeap.push(
+            std::make_pair(
+              perturbationScore(Ae, scores),
+              perturbationSets.size() - 1)
+            );
+      }
+
+    }while (! perturbationValid(Ai, numProj)  );//Discard invalid perturbations
+
+    // add perturbation vector to probing sequence if valid
+    for (size_t i = 0; i < Ai.size(); ++i)
+      additionalProbingBins(positions(Ai[i]), pvec) += actions(Ai[i]);
+
+  }
+  */
```

Comment:
If you like the new implementation, we can go ahead and remove the old one. :)  There are some other commented-out implementations above that we can remove too.